--- libobjc/exception.c	2013-02-03 11:16:21 +0000
+++ libobjc/exception.c	2013-02-03 11:16:21 +0000
@@ -202,7 +202,7 @@
 #ifdef SJLJ_EXCEPTIONS
 #define PERSONALITY_FUNCTION	__gnu_objc_personality_sj0
 #define __builtin_eh_return_data_regno(x) x
-#elif defined(__SEH__)
+#elif defined(__SEH__) && !defined (SJLJ_EXCEPTIONS)
 #define PERSONALITY_FUNCTION	__gnu_objc_personality_imp
 #else
 #define PERSONALITY_FUNCTION	__gnu_objc_personality_v0
@@ -227,7 +227,7 @@
 
 #define CONTINUE_UNWINDING return _URC_CONTINUE_UNWIND
 
-#ifdef __SEH__
+#if defined(__SEH__) && !defined (SJLJ_EXCEPTIONS)
 static
 #endif
 _Unwind_Reason_Code
@@ -524,7 +524,7 @@
   abort ();
 }
 
-#ifdef __SEH__
+#if defined(__SEH__) && !defined (SJLJ_EXCEPTIONS)
 EXCEPTION_DISPOSITION
 __gnu_objc_personality_seh0 (PEXCEPTION_RECORD ms_exc, void *this_frame,
 			     PCONTEXT ms_orig_context,
